### PR TITLE
Reduce number of plots resize observers

### PR DESCRIPTION
**Pull request by mattseddon**
Follow up from #5085

Reduces the number of `ResizeObserver`s that we need to 2 (1 for each plots grid).

### CODE REVIEW

**Review by julieg18 — APPROVED** [review]

File: webview/src/plots/components/customPlots/CustomPlots.tsx [hunk 1]
```diff
@@ -82,15 +83,15 @@ export const CustomPlots: React.FC<CustomPlotsProps> = ({ plotsIds }) => {
       onDragLeave={() => setOnSection(false)}
       onDragOver={handleDragOver}
       onDrop={handleDropAtTheEnd}
+      ref={gridRef}
     >
-      <Grid
-        setOrder={setPlotsIdsOrder}
+      <CustomPlotsGrid
+        gridRef={gridRef}
```
Comment (sroy3) [inline webview/src/plots/components/customPlots/CustomPlots.tsx:89]: Why does this need to be passed down? Can't the observer be initialized in here?

**Review by sroy3 — APPROVED** [review]

File: webview/src/plots/components/customPlots/CustomPlots.tsx (hunk 1 above)
Comment (mattseddon) [reply webview/src/plots/components/customPlots/CustomPlots.tsx:89]: discussed IRL.

tl;dr - ref must be passed down or the grid will appear to have 0 dimensions. The alternative is to use `useLayoutEffect`.

Comment (sroy3) [reply webview/src/plots/components/customPlots/CustomPlots.tsx:89]: I just tried it, and it works correctly without changing anything else. https://github.com/iterative/vscode-dvc/pull/5111
